h/migrations/versions/ff02f2c58_add_the_organization_table.py

```diff
@@ -0,0 +1,28 @@
+"""
+Add the organization table.
+
+Revision ID: ff02f2c58
+Revises: 7bf167611bc3
+Create Date: 2018-03-15 11:00:50.420618
+"""
+
+from __future__ import unicode_literals
+
+from alembic import op
+import sqlalchemy as sa
+
+revision = 'ff02f2c58'
+down_revision = '7bf167611bc3'
+
+
+def upgrade():
+    op.create_table(
+        'organization',
+        sa.Column('id', sa.Integer, autoincrement=True, primary_key=True),
+        sa.Column('name', sa.UnicodeText, nullable=False, index=True),
```

Comment:
Just wondering-why don't we use the same technique we used for group and have a public id?

Reply:
Yeah. The thing about pubids for groups is that they're necessary for _private_ groups, which were the only kind of group that we had when we first implemented groups and pubids:

* If I name my private group Foobar, and you're not a member of that group, then you don't even know that a group named Foobar exists. So we can't make people come up with unique names for their private groups, because that would reveal the names of other people's private groups.

* If you can guess the pubid of a private group then you can navigate to that group's page and then join the group. So the pubids need to be securely randomly generated.

But pubids actually suck because they make the URL of a group's page ugly, having this random string in it. For open and restricted groups pubids really aren't appropriate - at some point we probably want to get rid of them, and give those groups nicer URLs.

Anyway, so for organizations, you're right that having a human-chosen unique name forces the admin (or the developer, as it'll be for the time being) to do the work of coming up with a unique name. And given that these unique names aren't actually gonna be used anywhere user-visible right now (e.g. we're not gonna have organization URLs right away) they could just be randomly generated. But I think it's probably worth making the developer/admin do the work of coming up with a human readable one, so that organizations are ready to have nice URLs if/when we need them to?

And we should probably do the same for groups at some point as well, but that's down the road.

Reply:
I agree with you that we might as well do this now but I feel like this is just adding complexity by creating two separate ways of generating a pubid. It sounds like at some point in the future we want to change both these methods of generating a pubid anyway so i think it would be easier if they both just used the same method until such time as we change it. I see your point that they aren't pretty but I also think it's a pretty standard and elegant way of doing it and I rather not throw another method into the mix. Orgs and groups are very much the same animal to me and therefore it makes sense to me that they would use the same method to generate a pubid. Also who's to say we don't have private organizations at some point in the future? A final thought; could we simply make the name of the organization unique and avoid this?

Reply:
Ok, yeah I think just an `organization.pubid` the same as groups have sounds good, and then we can add human-readable unique names to both groups and organizations at the same time when the times comes, as you say. Even if we add human-readable unique names to both groups and organizations in the future, it'll still be good to have a generated public ID that's not the internal database pkey and that never changes.

I think you're right that orgs and groups have a _lot_ in common too

Reply:
I think I prefer adding the pubid over just making the name unique - I like the fact that the pubid never changes. Whereas a name might get edited over time (or if we have some sort of admin form for editing organizations one day, it would probably include changing the name.)